cmd/lotus-chainwatch/sync.go

```diff
@@ -302,40 +311,74 @@ func syncHead(ctx context.Context, api api.FullNode, st *storage, headTs *types.
 			alk.Unlock()
 		})
 
-		log.Infof("Getting miner info")
+		log.Infof("Getting actor change info")
 
 		minerChanges := 0
 		for addr, m := range actors {
 			for actor, c := range m {
-				if actor.Code != builtin.StorageMinerActorCodeID {
+				// reward actor
+				if actor.Code != builtin.RewardActorCodeID {
+					rewardTips[c.tsKey] = &rewardStateInfo{
+						stateroot:     c.stateroot,
+						baselinePower: big.Zero(),
+					}
 					continue
 				}
 
-				// only want miner actors with head change events
-				if _, found := headsSeen[actor.Head]; found {
-					continue
-				}
-				minerChanges++
+				// miner actors with head change events
+				if actor.Code != builtin.StorageMinerActorCodeID {
+					if _, found := headsSeen[actor.Head]; found {
+						continue
+					}
+					minerChanges++
 
-				minerTips[c.tsKey] = append(minerTips[c.tsKey], &minerStateInfo{
-					addr:      addr,
-					act:       actor,
-					stateroot: c.stateroot,
+					minerTips[c.tsKey] = append(minerTips[c.tsKey], &minerStateInfo{
+						addr:      addr,
+						act:       actor,
+						stateroot: c.stateroot,
 
-					tsKey:       c.tsKey,
-					parentTsKey: c.parentTsKey,
+						tsKey:       c.tsKey,
+						parentTsKey: c.parentTsKey,
 
-					state: miner.State{},
-					info:  miner.MinerInfo{},
+						state: miner.State{},
+						info:  miner.MinerInfo{},
 
-					rawPower: big.Zero(),
-					qalPower: big.Zero(),
-				})
+						rawPower: big.Zero(),
+						qalPower: big.Zero(),
+					})
 
-				headsSeen[actor.Head] = struct{}{}
+					headsSeen[actor.Head] = struct{}{}
+				}
+				continue
 			}
 		}
 
+		rewardProcessingStartedAt := time.Now()
+		parmap.Par(50, parmap.KVMapArr(rewardTips), func(it func() (types.TipSetKey, *rewardStateInfo)) {
+			tsKey, rewardInfo := it()
+			// get reward actor states at each tipset once for all updates
+			rewardActor, err := api.StateGetActor(ctx, builtin.RewardActorAddr, tsKey)
+			if err != nil {
+				log.Error(xerrors.Errorf("get reward state (@ %s): %w", rewardInfo.stateroot.String(), err))
+				return
+			}
+
+			rewardStateRaw, err := api.ChainReadObj(ctx, rewardActor.Head)
+			if err != nil {
+				log.Error(xerrors.Errorf("read state obj (@ %s): %w", rewardInfo.stateroot.String(), err))
+				return
+			}
+
+			var rewardActorState reward.State
+			if err := rewardActorState.UnmarshalCBOR(bytes.NewReader(rewardStateRaw)); err != nil {
+				log.Error(xerrors.Errorf("unmarshal state (@ %s): %w", rewardInfo.stateroot.String(), err))
+				return
+			}
+
+			rewardInfo.baselinePower = rewardActorState.BaselinePower
```

Comment:
BaselinePower will be disappearing with reward actor refactor: https://github.com/filecoin-project/specs-actors/pull/617/files#diff-15478070063dd16268193fab3a001125R11-R31

Reply:
My body is ready. :)